src/tunnel.cc

```diff
@@ -1021,6 +1027,15 @@ TunnelStateData::notePeerReadyToShovel(const Comm::ConnectionPointer &conn)
     }
 }
 
+void
+TunnelStateData::commitToServer(const Comm::ConnectionPointer &conn)
+{
+    committedToServer = true;
+    retriable = false; // may already be false
```

Comment:
Yes, but that can be viewed as a temporary coincidence, so I would rather not merge the two fields with rather different meanings, especially because it will make merging with FwdState more difficult. In the future, tunnels may retry more Squid-server errors, making `retriable` scope larger (crossing into the true `committedToServer` area) and also adding code that makes true committedToServer false again. Just like FwdState already does!